redis_json/src/c_api.rs

```diff
@@ -267,6 +272,54 @@ pub fn json_api_is_json<M: Manager>(m: M, key: *mut rawmod::RedisModuleKey) -> c
     m.is_json(key).map_or(0, |res| res as c_int)
 }
 
+pub fn json_api_get_key_value<'a, M: Manager>(_: M, val: *const c_void) -> *const c_void
+where
+    M::V: 'a,
+{
+    let json = unsafe { &*(val.cast::<M::V>()) };
+    match json.get_type() {
+        SelectValueType::Object => {
+            let vec = json.items().unwrap().collect::<Vec<(&'a str, &'a M::V)>>();
+            Box::into_raw(Box::new(KeyValuesIterator {
+                results: vec,
+                pos: 0,
+            }))
+            .cast::<c_void>()
+        }
+        _ => null(),
+    }
+}
+
+pub fn json_api_next_key_value<'a, M: Manager>(
+    _: M,
+    iter: *mut c_void,
+    ctx: *mut rawmod::RedisModuleCtx,
+    str: *mut *mut rawmod::RedisModuleString,
+) -> *const c_void
+where
+    M::V: 'a,
+{
+    let iter = unsafe { &mut *(iter.cast::<KeyValuesIterator<M::V>>()) };
+    if iter.pos < iter.results.len() {
+        create_rmstring(ctx, iter.results[iter.pos].0, str);
```

Comment:
I believe you can create a String with NULL context and so we can avoid the ctx parameter to the function. Also I believe we should check if `str` is not NULL so we will allow the user pass NULL  pointer as `str` indicating that he do not need the key.

Reply:
Regarding using NULL context - we already pass context for creating strings to other APIs,
So it would be inconsistent.
Agree on allowing `null` str when key name is not needed.

Reply:
So you prefer new APIs to avoid context? (and keep it for older APIs only which we cannot change)

Reply:
I believe so